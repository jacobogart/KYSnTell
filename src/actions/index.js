export const setLocations = (locations) => ({
  type: 'SET_LOCATIONS',
  locations
})