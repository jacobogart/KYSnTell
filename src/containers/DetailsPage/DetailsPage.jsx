import React, { Component } from 'react';
import { connect } from 'react-redux';

class DetailsPage extends Component {
  constructor(props) {
    super(props);
    this.state = {
      showContacts: false,
      diagnosis: '',
      timeFrame: '',
      additionalNotes: ''
    }
  }

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  toggleContacts = () => {
    this.setState({ showContacts: !this.state.showContacts})
  }
  
  render() {
    const contactList = this.props.contacts.map(contact => <li>{contact}</li>)
    const contactsDisplay = this.state.showContacts 
      ? <ol className="contact-list">
          {contactList}
        </ol>
      : <p onClick={this.toggleContacts}>{`${this.props.contacts[0]}...`}</p>
    return (
      <div className="DetailsPage">
        {contactsDisplay}
        <form classNall='details-form' onSubmit={this.handleSubmit}>
          <select
            name="timeFrame"
            onChange={this.handleChange}
            className="search-input"
          >
            <option value="10">10 miles</option>
            <option value="20">20 miles</option>
            <option value="30">30 miles</option>
            <option value="40">40 miles</option>
            <option value="50">50 miles</option>
          </select>
          <select
            name="distance"
            onChange={this.handleChange}
            className="search-input"
          >
            <option value="10">10 miles</option>
            <option value="20">20 miles</option>
            <option value="30">30 miles</option>
            <option value="40">40 miles</option>
            <option value="50">50 miles</option>
          </select>
          <input
            type="text"
            name="aditionalNotes"
            placeholder="Additional notes..."
            className="search-input"
            onChange={this.handleChange}
          />
          <button type="submit" >
            Use zipcode
        </button>
        </form>

      </div>
    );
  }
}

export const mapStateToProps = (state) => ({
  contacts: state.contacts
})

export default connect(mapStateToProps)(DetailsPage);