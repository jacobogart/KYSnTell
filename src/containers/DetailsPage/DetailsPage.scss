.DetailsPage {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  height: 78vh;
  margin-top: 15vh;
}

label,
p {
  color: white;
  font-size: 24px;
  margin: 5px;
}

.details-form {
  display: flex;
  flex-direction: column;
  justify-content: space-evenly;
  height: 60vh;
  width: 90vw;

  button,
  .search-input {
    background-color: white;
    font-size: 20px;
    padding: 8px;
    display: block;
    width: 90vw;
    border-radius: 10px;
  }

  .notes {
    width: 83vw;
  }
}

.contact-list-details {
  max-height: 150px;
  overflow: scroll;
}