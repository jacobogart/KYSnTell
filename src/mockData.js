import { locationCleaner } from "./api/cleaners";

export const origin = {
  lat: 38.9119463, 
  lng: -94.4136213 
}

export const destination = { 
  lat: "39.1952922", 
  lng: "-94.6817269" 
}

export const distance = '18.3 mi';

export const distanceResult = {
  rows: [
    {
      elements: [
        {
          distance: {
            text: distance
          }
        }
      ]
    }
  ]
}


export const fetchDistanceURL = "https://cors-anywhere.herokuapp.com/https://maps.googleapis.com/maps/api/distancematrix/json?origins=38.9119463,-94.4136213&destinations=39.1952922,-94.6817269&units=imperial&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const latLongResult = {
  results: [
    {
      geometry: {
        location: origin
      }
    }
  ]
}


export const fetchLatLongURL = "https://maps.googleapis.com/maps/api/geocode/json?address=64081&key=AIzaSyDHKnndIp7I8oymrxO0sDwpQUdZvfN1sqc"

export const locations = [
  {
    title: "Neighborhood Health",
    link: "https://gettested.cdc.gov/gettested_redirect/21952",
    id: "tag:hivtest.org,2000-03-20:/search/Detailx.asp?id=21952",
    streetAddress: "617 S 8th St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206-3819",
    telephone: "(615)-227-3000",
    categories: "",
    point: {
      lat: "36.1678930",
      long: "-86.7547047"
    },
    updated: "2019-02-08T17:19:32.0000000-05:00"
  },
  {
    title: "Street Works",
    link: "https://gettested.cdc.gov/gettested_redirect/110792",
    id: "tag:hivtest.org,2008-05-21:/search/Detailx.asp?id=110792",
    streetAddress: "520 Sylvan St",
    locality: "Nashville",
    region: "TN",
    postalCode: "37206",
    telephone: "(615)-259-7676",
    categories: "",
    point: {
      lat: "36.1665114",
      long: "-86.7605114"
    },
    updated: "2018-08-02T18:05:30.0000000-04:00"
  }
];

export const locationsResult = {
  services: [
    {
      providers: locations
    }
  ]
}

export const cleanLocations = locations.map(location => locationCleaner(location))

export const fetchLocationsURL = "https://cors-anywhere.herokuapp.com/https://locator.hiv.gov/data?lat=38.9119463&long=-94.4136213&distance=10"

export const contact = '+18168682142';

export const contacts = [contact, contact]

export const message = 'Test message.'

export const sendMessageOptions = {
  method: 'POST',
  headers: {
    'Content-Type': 'application/json'
  },
  body: JSON.stringify({ to: contact, message })
}

export const resolvedMessages = [{ ok: true }, { ok: true }];

export const cleanLocation = {
  title: 'Neighborhood Health',
  link: 'https://gettested.cdc.gov/gettested_redirect/21952',
  id: '21952',
  address: '617 S 8th St;Nashville, TN 37206-3819',
  telephone: '(615)-227-3000',
  point: { lat: '36.1678930', lng: '-86.7547047' }
}

export const details = {
  additionalNotes: "Test",
  diagnosis: "Herpes",
  timeFrame: "year"
}

export const user = {
  location: origin
}

export const searchChangeEvent = {
  target: {
    name: 'zipcode',
    value: '64081'
  }
}

export const submitEvent = {
  preventDefault: jest.fn()
}

export const history = {
  push: jest.fn()
}

export const setUserLocation = jest.fn();

export const setLocations = jest.fn();

export const setContacts = jest.fn();

export const fullMessage = 'Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider. Additional notes from partner: Test'

export const messageWONotes = "Hello, we have been informed by an anonymous sexual partner that you may have been exposed to Herpes in the last year. While this is no cause for alarm, we do recommend getting tested at your earliest convenience. To find a testing center near you, please visit kysntell.com, or contact your preferred healthcare provider."

export const previewEvent = {
  target: {
    blur: jest.fn()
  }
}

export const messageContext = {
  focus: jest.fn(),
  blur: jest.fn(),
  innerText: 'Test'
}

 export const rejectedMessages = [{ ok: false, contact }]

 export const routerLocation = {
  pathname:'/kys/locations/21952'
 }

export const detailsChangeEvent = {
  target: {
    name: 'diagnosis',
    value: 'Herpes'
  }
}

export const setDetails = jest.fn();

export const contactChangeEvent = {
  target: {
    value: contact
  }
}

export const contactBadChangeEvent = {
  target: {
    value: 'test'
  }
}

export const searchDefaultState = {
  zipcode: '',
  distance: '',
  isLoading: false,
  error: false
}