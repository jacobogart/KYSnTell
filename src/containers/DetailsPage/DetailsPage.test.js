import React from 'react';
import { shallow } from 'enzyme';
import { DetailsPage, mapStateToProps, mapDispatchToProps } from './DetailsPage';
import * as mock from '../../mockData'
import { setDetails } from '../../actions';

describe('DetailsPage', () => {
  let wrapper, instance;

  beforeEach(() => {
    wrapper = shallow(
      <DetailsPage 
        contacts={mock.contacts}
        setDetails={mock.setDetails}
        history={mock.history}
      />);
    instance = wrapper.instance();
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should have default state', () => {
    const defaultState = {
      diagnosis: '',
      timeFrame: '',
      additionalNotes: ''
    }
    expect(wrapper.state()).toEqual(defaultState);
  });

  describe('Event Listeners', () => {
    it('should invoke handleSubmit when the form is submited', () => {
      wrapper.find('.details-form').simulate('submit', mock.submitEvent);
      expect(mock.submitEvent.preventDefault).toHaveBeenCalled();
    });

    it('should invoke handleChange when an input is changed', () => {
      wrapper.find('#diagnosis').simulate('change', mock.detailsChangeEvent);
      expect(wrapper.state('diagnosis')).toEqual('Herpes');
    });
  });

  describe('handleChange', () => {
    it('should set the correct piece of state to the correct value', () => {
      instance.handleChange(mock.detailsChangeEvent);
      expect(wrapper.state('diagnosis')).toEqual('Herpes');
    });
  });

  describe('handleSubmit', () => {
    it('should invoke setDetails', () => {
      instance.setState(mock.details);
      instance.handleSubmit(mock.submitEvent);
      expect(mock.setDetails).toHaveBeenCalledWith(mock.details);
    });

    it('should push the router to the correct URL', () => {
      instance.handleSubmit(mock.submitEvent);
      expect(mock.history.push).toHaveBeenCalledWith('/tell/preview')
    });
  });

  describe('mapStateToProps', () => {
    it('should return the correct piece of state', () => {
      const { contacts, user } = mock;
      const defaultState = { contacts, user };
      const result = mapStateToProps(defaultState);
      expect(result).toEqual({ contacts })
    });
  });

  describe('mapDispatchToProps', () => {
    it('should dispatch a setDetails action', () => {
      const mockDispatch = jest.fn();
      const mappedProps = mapDispatchToProps(mockDispatch);
      const actionToDispatch = setDetails(mock.details);
      mappedProps.setDetails(mock.details);
      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});