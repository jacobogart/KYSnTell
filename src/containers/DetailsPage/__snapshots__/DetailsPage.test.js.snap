// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailsPage should match snapshot 1`] = `
<div
  className="DetailsPage"
>
  <p>
    Contacts
  </p>
  <ol
    className="contact-list-details contact-list"
  >
    <li
      key="+18168682142"
    >
      (816) 868-2142
    </li>
    <li
      key="+18168682142"
    >
      (816) 868-2142
    </li>
  </ol>
  <form
    className="details-form"
    onSubmit={[Function]}
  >
    <div
      className="select-holder"
    >
      <label
        htmlFor="diagnosis"
      >
        Diagnosis
      </label>
      <select
        className="search-input"
        id="diagnosis"
        name="diagnosis"
        onChange={[Function]}
      >
        <option
          key="Select"
          value="Select"
        >
          Select
        </option>
        <option
          key="HIV/AIDS"
          value="HIV/AIDS"
        >
          HIV/AIDS
        </option>
        <option
          key="HPV(Human Papillomavirus)"
          value="HPV(Human Papillomavirus)"
        >
          HPV(Human Papillomavirus)
        </option>
        <option
          key="Chlamydia"
          value="Chlamydia"
        >
          Chlamydia
        </option>
        <option
          key="Gonorrhea"
          value="Gonorrhea"
        >
          Gonorrhea
        </option>
        <option
          key="Syphilis"
          value="Syphilis"
        >
          Syphilis
        </option>
        <option
          key="Herpes"
          value="Herpes"
        >
          Herpes
        </option>
        <option
          key="Trichomoniasis"
          value="Trichomoniasis"
        >
          Trichomoniasis
        </option>
      </select>
    </div>
    <div
      className="select-holder"
    >
      <label
        htmlFor="timeFrame"
      >
        Time since last test
      </label>
      <select
        className="search-input"
        id="timeFrame"
        name="timeFrame"
        onChange={[Function]}
      >
        <option
          value=""
        >
          Select
        </option>
        <option
          value="month"
        >
          one month
        </option>
        <option
          value="two months"
        >
          two months
        </option>
        <option
          value="three months"
        >
          three months
        </option>
        <option
          value="six months"
        >
          six months
        </option>
        <option
          value="year"
        >
          year
        </option>
        <option
          value="other"
        >
          other
        </option>
      </select>
    </div>
    <input
      className="search-input notes"
      name="additionalNotes"
      onChange={[Function]}
      placeholder="Additional notes..."
      type="text"
    />
    <button
      type="submit"
    >
      Preview
    </button>
  </form>
</div>
`;
